framework/source/class/qx/log/Logger.js

```diff
@@ -578,7 +578,12 @@ qx.Bootstrap.define("qx.log.Logger",
       // Add relation fields
       if (object)
       {
-        function isQxCoreObject(clz) {
+        // Do not explicitly check for instanceof qx.core.Object, in order not
+        // to introduce an unwanted load-time dependency
+        function isQxCoreObject(object) {
+          if (object === object.constructor)
+            return false;
```

Comment:
I am afraid this, again, won't be accepted by linting without braces.... :-(

Reply:
:sigh: IMHO lint is 99% vanity and 1% sanity, this rule is part of the 99%...

Reply:
We can change the rules - You could create an issue here (I think): https://github.com/qooxdoo/eslint-config-qx

Reply:
I'm one of the ones who feels strongly that this lint rule is part of the 1% (and I' think I give it more of a 50/50 spread rather than 99/1 :-) ), because it prevents accidentally adding a new statement, e.g. a console.log inserted at line 584.5, and causing a complete change of operation (the return statement is then not part of the `if`). There are occasional times when such things are useful, though. I would be fine with enabling the lint option that allows a single line without braces, so this would end up looking like:

```
    if (object == object.constructor) return false;
```

Reply:
I think that it depends a lot on the context, and for me it's all about readability and the flow of the code.  For example, enforcing indentation is a good solution for the problem of accidental new statements and preserves readability, so having a single line like that increases the chances of errors, eg

```
  if (object === object.constructor) return false;
    console.log("it's a constructor");
```

I find that I read the first token on the line plus the indentation to see when to skip ahead, and so one line statements are dangerous and time consuming because you have to read to the end.

Consistent indentation is really key, which to me means that this is troublesome:

```
    if (abc.getSomeReallyLong().methodName() == my.package.name.here.SomeClass.VALUE &&
        yetMoreCode.getBlahDeBlah() == my.package.name.here.SomeClass.OTHER_VALUE)
      return true;
```

That's not ideal IMHO, and should have brackets around `return true` - possibly on separate lines, a practice I usually detest because it introduces yet another extra bit of vertical cruft.

OTOH I find that the brackets rule is ugly for many simple statements, elongates the code and just gets in the way of reading the meaning of the code.

```
  if (a){
    doSomething(1);
  }
  if (b) {
    doSomething(2);
  }
```
as opposed to:
```
if (a)
  doSomething(1);
if (b)
  doSomething(2);
```

Ultimately though, these debates just go to show that that lint is highly subjective.  For me, I find lint occasionally useful, just not very often - it is particularly useless when it is only visible when I commit and travis kicks in, and pretty much the most it can possibly do at that stage is show up a code path which breaks and for which there is no unit test.

I'm not proposing changing the lint rules BTW, just having a moan 😆  It would be much better if (when) we could one day integrate lint on the fly with the continuous compilation of `qx compile` *and* present that somewhere front and centre (i.e somewhere other than in a terminal window that you never look at until it's too late...)